### PR TITLE
CLC-6062: fix broken Selenium tests and speed up QA builds

### PR DESCRIPTION
**Pull request by pfarestveit**
https://jira.ets.berkeley.edu/jira/browse/CLC-6062

Misc test fixes.

### CODE REVIEW

File: spec/ui_selenium/my_academics_data_advising_appts_spec.rb [hunk 1]
```diff
@@ -1,6 +1,6 @@
 describe 'My Academics L&S Advising card', :testui => true do
 
-  if ENV["UI_TEST"]
+  if ENV["UI_TEST"] && Settings.ui_selenium.layer != 'production'
```
Comment (johncrossman) [inline spec/ui_selenium/my_academics_data_advising_appts_spec.rb]: @pfarestveit If this is used often you could easily move to a module util. Future enhancement, perhaps.

File: spec/ui_selenium/my_dashboard_canvas_course_sites_spec.rb [hunk 2]
```diff
@@ -1,307 +1,311 @@
...
-
-        @canvas.log_out @cal_net
-
-        # Wait for Canvas to index new data
-        sleep WebDriverUtils.canvas_update_timeout
-
-        # Clear cache so that new data will load immediately
-        UserUtils.clear_cache(@driver, @splash_page, @my_dashboard)
-      end
+  if ENV["UI_TEST"]
```
Comment (johncrossman) [inline spec/ui_selenium/my_dashboard_canvas_course_sites_spec.rb]: Can this be added to the `describe 'My Dashboard', :testui => true, :order => :defined do` statement?

Comment (pfarestveit) [inline spec/ui_selenium/my_dashboard_canvas_course_sites_spec.rb]: Thanks @johncrossman.  I'll put that in the "refactor tests" list.

**Comment by johncrossman** [comment]
50% of Travises are happy and I need to go get a drink

**Pull request merged**